test/Swashbuckle.AspNetCore.SwaggerGen.Test/SchemaGenerator/JsonSerializerSchemaGeneratorTests.cs

```diff
@@ -770,6 +770,17 @@ public void GenerateSchema_HonorsSerializerOption_PropertyNamingPolicy()
             Assert.Equal(expectedDefaultAsJson, propertySchema.Default.ToJson());
         }
 
+        [Fact]
+        public void GenerateSchema_HonorsEnumDictionaryKeys_StringEnumConverter()
+        {
+            var subject = Subject();
+            var schemaRepository = new SchemaRepository();
+
+            var referenceSchema = subject.GenerateSchema(typeof(Dictionary<IntEnum, string>), schemaRepository);
+
+            Assert.Equal(typeof(IntEnum).GetEnumValues().Length, referenceSchema.Properties.Count);
```

Comment:
This doesn't seem to have an assertion that's integer specific - wouldn't this assertion pass regardless of whether the keys were integers or strings as the number of properties would be the same in both cases?

Reply:
e.g.

{ enumValue1: string, enumValue2: string ...} vs
{ [key: string]: string }

Reply:
In that case, can we assert that the properties are all of type string and have the correct names for that enum?

Reply:
Sure, give me a sec so I compare the names. Maybe not the worst idea so I can check if `options.JsonSerializerOptions.DictionaryKeyPolicy` is respected.